packages/main/src/plugin/authentication.ts

```diff
@@ -0,0 +1,189 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type {
+  authentication,
+  AuthenticationProvider,
+  AuthenticationSession,
+  AuthenticationSessionsChangeEvent,
+  AuthenticationGetSessionOptions,
+  Event,
+  AuthenticationProviderAuthenticationSessionsChangeEvent,
+  AuthenticationSessionAccountInformation,
+  AuthenticationProviderOptions,
+  Disposable,
+} from '@podman-desktop/api';
+// import { window } from '@podman-desktop/api';
+import { Emitter } from './events/emitter';
+type Authentication = typeof authentication;
+
+/**
+ * Structure to save authentication provider information
+ * with additional metadata
+ */
+export interface ProviderWithMetadata {
+  id: string;
+  label: string;
+  provider: AuthenticationProvider;
+  options: AuthenticationProviderOptions;
+}
+
+export interface AuthenticationProviderInfo {
+  id: string;
+  displayName: string;
+  accounts: AuthenticationSessionAccountInformation[];
+}
+
+export interface ExtensionInfo {
+  id: string;
+  label: string;
+}
+
+export interface AllowedExtension {
+  id: string;
+  name: string;
+  allowed?: boolean;
+}
+
+export class AuthenticationImpl {
+  private _authenticationProviders: Map<string, ProviderWithMetadata> = new Map<string, ProviderWithMetadata>();
+
+  constructor(private apiSender: any) {}
+
+  public async getAuthenticationProvidersInfo(): Promise<AuthenticationProviderInfo[]> {
+    const values = Array.from(this._authenticationProviders.values());
+    const sessionsRequests = values.map(meta => {
+      return meta.provider.getSessions().then(sessions => {
+        return {
+          id: meta.id,
+          displayName: meta.label,
+          accounts: sessions.map(session => ({ id: session.id, label: session.account.label })),
+        };
+      });
+    });
+
+    return await Promise.all(sessionsRequests);
+  }
+
+  public async signOut(providerId: string, sessionId: string) {
+    this.removeSession(providerId, sessionId);
+  }
+
+  registerAuthenticationProvider(
+    id: string,
+    label: string,
+    provider: AuthenticationProvider,
+    options?: AuthenticationProviderOptions,
+  ): Disposable {
+    if (this._authenticationProviders.get(id)) {
+      throw new Error(`An authentication provider with id '${id}' is already registered.`);
+    }
+    this._authenticationProviders.set(id, {
+      id,
+      label,
+      provider,
+      options: options ?? { supportsMultipleAccounts: false },
+    });
+    this.apiSender.send('authentication-provider-update');
```

Comment:
should it provide the id as argument ?
```suggestion
    this.apiSender.send('authentication-provider-update', {id});
```

Reply:
it can, but right now UI just pulls info about all providers/sessions when gets this event.

Reply:
yes, it's fine on UI side for now, just that 'core' can send more data so we could be better at the UI side if we want